os_migrate/plugins/module_utils/keypair.py

```diff
@@ -29,7 +29,11 @@ class Keypair(resource.Resource):
 
     @staticmethod
     def _create_sdk_res(conn, sdk_params):
-        return conn.compute.create_keypair(**sdk_params)
+        try:
+            return conn.compute.create_keypair(**sdk_params)
+        except openstack.exceptions.BadRequestException:
+            sdk_params_no_type = {k: v for k, v in sdk_params.items() if k != 'type'}
```

Comment:
Would using `del` be a bit cleaner?

```
if 'type' in sdk_params.keys():
    # With BadRequestException it is likley that 'type' is not yet supported. Remove and try again
    del sdk_params['type']
```

Reply:
Thank you for the reasoning. I couldn't immediately see any other use of `sdk_params` but that doesn't mean it won't happen in the future. +1 on your change.